launcher/minecraft/auth/AccountList.cpp

```diff
@@ -283,9 +283,14 @@ QVariant AccountList::data(const QModelIndex& index, int role) const
                     return account->accountDisplayString();
 
                 case TypeColumn: {
-                    auto typeStr = account->typeString();
-                    typeStr[0] = typeStr[0].toUpper();
-                    return typeStr;
+                    switch (account->accountType()) {
+                        case AccountType::MSA: {
+                            return tr("MSA", "Account type");
+                        }
+                        case AccountType::Offline: {
+                            return tr("Offline", "Account type");
+                        }
```

Comment:
Here add a default case for the "unknown" or better use the old approach but just wrap it around `tr`:
```
auto typeStr = account->typeString();
typeStr[0] = typeStr[0].toUpper();
return tr(typeStr,"Account type");
```

Reply:
I am very sure that `tr(<some dynamic variable>)` won't work, as the Qt's translation tooling only looks at the code statically

Reply:
Then just having a default case would suffice

Reply:
The `AccountType` enum only has 2 values: `MSA` and `Offline`. I'm confused on why there is code at all (`MinecraftAccount::typeString()`, for example) that switches on the value of an `AccountType` variable, covers all the cases, yet still has a `default:` block.

Reply:
It's mostly for future proofing. If a future contributor forgets to add a case after adding another enum value, there would be some defined behavior

Reply:
On the contrary, I think it's actively harmful for future proofing.

Compilers have warnings that flag switches that don't cover all the values of an enum: `-Wswitch` on GCC and Clang, enabled already by `-Wall`, and C4062 on MSVC, enabled already by `/W4`. If someone adds a new enum value, they will get warnings about not handling it. These warnings will give them a clear list of places they need to handle it.

But with the current situation, if someone adds a new enum value, they have no idea where they need to add new code to handle it. If someone adds a `Foo` account type, it should *not* be shown as "unknown", but as "Foo". But they could very easily miss the fact that they need to change some random bit of UI code to handle the new account type.

Reply:
I agree, it's probably best to have a return at the bottom though as I don't think that silences the warning

Reply:
> But with the current situation, if someone adds a new enum value, they have no idea where they need to add new code to handle it

Agreed. Let's keep this as-is then.

Reply:
I'm okay with a return "unknown" at the bottom, that way the warning is still maintained, and in the unlike case that the warning is ignored have an exit condition.

Reply:
> in the unlike case that the warning is ignored

We could add `-Werror` to the compilation flags to make this impossible.

Reply:
Do whatever you like. 
But I can't understand why can't you just add a return statement outside the switch statement.

Reply:
I'm fine with adding it if you really want it, but I'm proposing other ideas because I think we can achieve a more optimal solution. As I understand, you want it to be there because you worry that a case might not be handled at some point in the future. I want to remove that worry once and for all, to make that error impossible, and, in the process, make all of these just-in-case `default`/`return`s unnecessary.

Reply:
> > in the unlike case that the warning is ignored
> 
> We could add `-Werror` to the compilation flags to make ignoring it impossible (or selectively add `-Werror=switch`).

Hard no on `Werror` as we have lots of different downstream environments that might will use different compilers and -versions, so it's basically impossible to ensure that there are zero warnings at all times

Reply:
Okay, I've added `-Werror=switch` and `/we4062` to the build options. Instead of returning "unknown", I've marked the code after the switch with `unreachable()`; I think that better communicates the intent here. I can't use `std::unreachable()` because that's only a C++23 feature, so I've included a short custom definition of it.

Reply:
First didn't you mention that that error is already activated by `-Wall` and `/W4`? If yes just revert the changes from `cmake/CompilerWarnings.cmake` the mentioned two flags are already added.
Second, you do not need to define an `unreachable` function just use `Q_ASSERT(false);` with a fancy comment.
Even so, I would still prefer a return there as it's easier and much cleaner from my perspective.

Reply:
Alright, I changed it to a return "unknown". `-Wall` and `/W4` enable the warning, but you expressed the concern that people might ignore the warning, so the `cmake/CompilerWarnings.cmake` changes promote them to errors so ignoring them is impossible.

Reply:
@Scrumplex how does the cmake changes look to you?
To me, they seem useless (as they should be already included in a different flag).
Are they ok?

Reply:
i don't believe they were errors before